### DISCUSSION TITLE
Question regarding upstream order Adguard Home chooses to use.

### DISCUSSION THREAD

**Discussion by emlimap** [AdguardTeam/AdGuardHome#3464] [Q&A] [answered]
I am trying to understand how AdGuard Home handles when there is an upstream config for a specific subdomain as well as one at domain level for same domain.

From what I have seen, AdGuard Home chooses to use the domain level upstream instead of specific domain. In below example, all DNS queries for `example.akamaized.net` goes to Google DNS instead of `1.2.3.4`. Ordering in the config file doesn't seem to make a difference either. Is this expected behaviour?

```
[/example.akamaized.net/]1.2.3.4
[/akamaized.net/akamaihd.net/akamai.net/akamaiedge.net/akadns.net/akahost.net/edgekey.net/]tls://dns.google

**Comment by ainar-g** [comment, marked as answer]
We cannot reproduce this.  In fact, the [documentation] clearly says that the most specific subdomain is used.  Please make sure that your real domain is spelled correctly.

[documentation]: https://github.com/AdguardTeam/AdGuardHome/wiki/Configuration#upstreams-for-domains

**Reply by emlimap** [reply]
Managed to track it down, and it turns out that it was because of client specific override I had set a while ago and forgot about it. So global upstream DNS servers where I had the config were being ignored. Apologies for the confusion.
